cob_twist_controller/CMakeLists.txt

```diff
@@ -28,10 +28,11 @@ set(SRC_C_DIR "src/constraint_solvers")
 set(SRC_CS_DIR "${SRC_C_DIR}/solvers")
 
 # Declare a cpp library
-add_library(damping_methods src/damping_methods/damping_base.cpp src/damping_methods/damping.cpp)
-add_library(constraint_solvers ${SRC_CS_DIR}/constraint_solver_base.cpp ${SRC_CS_DIR}/unconstraint_solver.cpp ${SRC_CS_DIR}/wln_joint_limit_avoidance_solver.cpp ${SRC_CS_DIR}/weighted_least_norm_solver.cpp ${SRC_C_DIR}/constraint_solver_factory_builder.cpp)
-add_library(limiters src/limiters/limiter_base.cpp src/limiters/limiter.cpp)
-target_link_libraries(constraint_solvers damping_methods)
+add_library(damping_methods src/damping_methods/damping.cpp)
+add_library(constraint_solvers ${SRC_CS_DIR}/unconstraint_solver.cpp ${SRC_CS_DIR}/wln_joint_limit_avoidance_solver.cpp ${SRC_CS_DIR}/weighted_least_norm_solver.cpp ${SRC_C_DIR}/constraint_solver_factory_builder.cpp)
+add_library(limiters src/limiters/limiter.cpp)
+add_library(pseudoinv_calculations src/pseudoinverse_calculations/pseudoinverse_calculation.cpp)
```

Comment:
The pseudo-inverse is the inverse that is returned by SVD (see Nakamura Y., Advanced Robotics: Redundancy and Optimization, p. 34). Furthermore the singular values are damped. filtered in this base in the current solution. So the name is good for the current solution but not for the general cased as mentioned in your comment. 

Therefore I am going to rename it.